pymc_extras/statespace/models/structural/components/cycle.py

```diff
@@ -0,0 +1,307 @@
+            k_endog=k_endog,
+            k_states=k_states,
+            k_posdef=k_posdef,
+            measurement_error=False,
+            combine_hidden_states=True,
+            obs_state_idxs=obs_state_idx,
+            observed_state_names=observed_state_names,
+        )
+
+    def make_symbolic_graph(self) -> None:
+        if self.k_endog == 1:
+            self.ssm["design", 0, slice(0, self.k_states, 2)] = 1
+            self.ssm["selection", :, :] = np.eye(self.k_states)
+            init_state = self.make_and_register_variable(f"{self.name}", shape=(self.k_states,))
+
+        else:
+            Z = np.array([1.0, 0.0]).reshape((1, -1))
+            design_matrix = linalg.block_diag(*[Z for _ in range(self.k_endog)])
+            self.ssm["design", :, :] = pt.as_tensor_variable(design_matrix)
+
+            R = np.eye(2)  # 2x2 identity for each cycle component
+            selection_matrix = linalg.block_diag(*[R for _ in range(self.k_endog)])
+            self.ssm["selection", :, :] = pt.as_tensor_variable(selection_matrix)
+
+            init_state = self.make_and_register_variable(f"{self.name}", shape=(self.k_endog, 2))
+
+        self.ssm["initial_state", :] = init_state.ravel()
+
+        if self.estimate_cycle_length:
+            lamb = self.make_and_register_variable(f"{self.name}_length", shape=())
+        else:
+            lamb = self.cycle_length
+
+        if self.dampen:
+            rho = self.make_and_register_variable(f"{self.name}_dampening_factor", shape=())
+        else:
+            rho = 1
+
+        T = rho * _frequency_transition_block(lamb, j=1)
+        if self.k_endog == 1:
+            self.ssm["transition", :, :] = T
+        else:
+            transition = block_diag(*[T for _ in range(self.k_endog)])
+            self.ssm["transition"] = pt.specify_shape(transition, (self.k_states, self.k_states))
+
+        if self.innovations:
+            if self.k_endog == 1:
+                sigma_cycle = self.make_and_register_variable(f"sigma_{self.name}", shape=())
+                self.ssm["state_cov", :, :] = pt.eye(self.k_posdef) * sigma_cycle**2
+            else:
+                sigma_cycle = self.make_and_register_variable(
+                    f"sigma_{self.name}", shape=(self.k_endog,)
+                )
+                state_cov = block_diag(
+                    *[pt.eye(2) * sigma_cycle[i] ** 2 for i in range(self.k_endog)]
+                )
+                self.ssm["state_cov"] = pt.specify_shape(state_cov, (self.k_states, self.k_states))
+
+    def populate_component_properties(self):
+        if self.k_endog == 1:
+            self.state_names = [f"{self.name}_{f}" for f in ["Cos", "Sin"]]
+        else:
+            # For multivariate cycles, create state names for each observed state
+            self.state_names = []
+            for var_name in self.observed_state_names:
+                self.state_names.extend([f"{self.name}_{var_name}_{f}" for f in ["Cos", "Sin"]])
```

Comment:
Let's use a square brackets notation here, so the states will be e.g. `business_cycle_Cos[data_1]`, `business_cycle_Sin[data_1]`, etc.

```
self.state_names = [f'{name}_{f}[{var_name}]' if self.k_endog > 1 else f'{name}_{f}'
               for var_name in self.observed_state_names
               for f in ["Cos", "Sin"] 
]
```